src/components/Inngest.ts

```diff
@@ -163,6 +166,13 @@ export class Inngest<TOpts extends ClientOptions = ClientOptions> {
       inngestEnv: env,
     });
 
+    const signingKey = processEnv(envKeys.SigningKey) || "";
```

Comment:
This is tricky. Some environments don't give you access to environment variables upon instantiation and instead pass them _alongside a request_, e.g. Cloudflare.

For this reason, `InngestCommHandler` attempts to pull these [as late as possible](https://github.com/inngest/inngest-js/blob/main/src/components/InngestCommHandler.ts#L646).

To support these types of environments, it might be reasonable for `upsertSigningKeyFromEnv` to _also_ update `this.client["inngestApi"]` (maybe via a `this.client["inngestApi"].setSigningKey()` method) with the new key as well as setting it for itself.